### DISCUSSION TITLE
Styling Accordion container based on expanded state

### DISCUSSION THREAD

**Discussion by bhainesva** [chakra-ui/chakra-ui#8406] [Q&A] [answered]
Hello, I'd like to style an Accordion's `container` part depending on whether the item is open or not. 

It seems like the _expanded state is only passed to the `button`, so my initial thought of putting this in my theme didn't work: 
```js
const components = {
  Accordion: {
    baseStyle: definePartsStyle({
      container: {
        border: '1px solid orange',
        _expanded: {
          border: '1px solid green',
        },
      },
    })
  }
}
```

My next thought was that I could use a  custom class and do something like this: 
```js
const components = {
  Accordion: {
    baseStyle: definePartsStyle({
      container: {
        border: '1px solid orange',
       '&.is-expanded': {
          border: '1px solid green',
        },
      },
    })
  }
}
```
But I don't know an ergonomic way to get the `is-expanded` class onto the container. I could do something like:
```jsx
<Accordion>
  <AccordionItem>
   ({ isExpanded }) => (
    <chakra.div className="is-expanded">
      // content...
    </chakra.div>
  )
  </AccordionItem>
</Accordion>
```

But I don't want to require users to do this themselves. I thought about implementing my own AccordionItem by copying and pasting the Chakra implementation but some of the required pieces like `AccordionItemProvider` aren't exported. 

The answer I've currently landed on is this, but is there a nicer way to accomplish this?:
```jsx
// AccordionItem.tsx
import { chakra, AccordionItem as ChakraAccordionItem, AccordionItemProps } from "@chakra-ui/react";

export function AccordionItem(props: AccordionItemProps) {
  const { children, ...rest } = props;

  return (
    <ChakraAccordionItem {...rest}>
      {({ isExpanded, isDisabled }) => (
        <chakra.div className={isExpanded ? 'is-expanded' : undefined}>
          {typeof children === 'function'
            ? children({ isExpanded, isDisabled })
            : children
          }
        </chakra.div>
      )}
    </ChakraAccordionItem>
  )
}

// theme.ts
const components = {
  Accordion: {
    baseStyle: definePartsStyle({
      container: {
        '& > *': {
          border: '1px solid orange',
        },
        '& > .is-expanded': {
          border: '1px solid green',
        }
      },
    })
  }
}
```

**Comment by marqpdx** [comment]
Yeah am bumping against the same: `_expanded` doesn't appear to work in a theme file, unfortunately. thanks for the above solution!

**Comment by SrckyImpact** [comment]
I managed to achieve this by utilizing CSS `:has` selector:
```
    <AccordionItem
      borderWidth={'1px'}
      sx={{
        ':has(button[aria-expanded="true"])': {
          borderColor: 'foreground-brand-primary',
        },
      }}
    >
```

**Comment by segunadebayo** [comment, marked as answer]
In v3, Zag sets `data-state="open"` on `Accordion.Item`. Style it with `_open`.

```tsx
<Accordion.Item value="a" _open={{ borderColor: "green" }} />
```

Same thing in the accordion recipe: `item: { _open: { borderColor: "green" } }`.
